tests/Unit/Elliptic/DiscontinuousGalerkin/Test_DgOperator.cpp

```diff
@@ -253,6 +253,91 @@ struct Metavariables {
   void pup(PUP::er& /*p*/) {}
 };
 
+template <size_t Dim>
+struct ModifiedPoissonSystem
+    : Poisson::FirstOrderSystem<Dim, Poisson::Geometry::FlatCartesian> {
+  struct modify_boundary_data {
+    using argument_tags = tmpl::list<
+        domain::Tags::Element<Dim>,
+        ::Tags::Mortars<domain::Tags::Coordinates<Dim, Frame::Inertial>, Dim>>;
+    static void apply(
+        const gsl::not_null<Scalar<DataVector>*> field,
+        const gsl::not_null<Scalar<DataVector>*> normal_dot_flux,
+        const DirectionalId<Dim>& mortar_id, const Element<Dim>& element,
+        const DirectionalIdMap<Dim, tnsr::I<DataVector, Dim, Frame::Inertial>>&
+            all_mortar_coords) {
+      // Assuming that in the region x > 0.5 (block 1) we decompose the field as
+      // u = u_R + u_P with u_P = 2x + 3y
+      const auto& element_id = element.id();
+      const bool element_is_modified = element_id.block_id() == 1;
+      const bool neighbor_is_modified = mortar_id.id().block_id() == 1;
+      if (element_is_modified == neighbor_is_modified) {
+        return;
+      }
+      const auto& x = all_mortar_coords.at(mortar_id);
+      const DataVector singular_field = 2. * get<0>(x) + 3. * get<1>(x);
+      const auto& direction = mortar_id.direction();
+      // Assuming rectilinear grid to simplify the normal vector
+      const double singular_normal_dot_flux =
+          (direction.dimension() == 0 ? 2. : 3.) * direction.sign();
+      // In modified elements, the field is u_R = u - u_P
+      const double sign = element_is_modified ? -1. : 1.;
+      get(*field) += sign * singular_field;
+      // Minus sign because we are modifying the _received_ fluxes, which were
+      // computed using the neighbor's face normal
+      get(*normal_dot_flux) -= sign * singular_normal_dot_flux;
+    }
+  };
+};
+
+template <size_t Dim>
+struct ModifiedPoissonSolution : Poisson::Solutions::ProductOfSinusoids<Dim> {
+  using Poisson::Solutions::ProductOfSinusoids<Dim>::ProductOfSinusoids;
+  std::unique_ptr<elliptic::analytic_data::AnalyticSolution> get_clone()
+      const override {
+    return std::make_unique<ModifiedPoissonSolution>(*this);
+  }
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(ModifiedPoissonSolution);  // NOLINT
+  template <typename... RequestedTags>
+
+  tuples::TaggedTuple<RequestedTags...> variables(
```

Comment:
Formatting.  I was looking all over for where `RequestedTags` was defined.

Reply:
fixed